services/src/main/java/org/fao/geonet/api/registries/vocabularies/KeywordsApi.java

```diff
@@ -443,6 +443,11 @@ public Element getKeywordById(
                 requestParams.addContent(new Element(e.getKey()).setText(e.getValue()));
             }
         }
+
```

Comment:
It was not added by line 443 which loops on allRequestParams ?

Reply:
Indeed, I got a bit confused with the explicit parameter defined in the service not being used, but the loop indeed adds the  parameter.

What is the reason to use `allRequestParams`? do we need to send dynamic parameters? In that case I think should be better to remove the explicit parameters from the service API. 

But otherwise, if no need to send  dynamic parameters I think we should avoid the usage `allRequestParams` and define the explicit parameters of the service.

Reply:
I have a project where the client side inject some more parameters depending on the editor (with 19115-3).

Reply:
Reverted to previous code